enzy_htp/structure/structure_constraint/api.py

```diff
@@ -155,6 +166,13 @@ def check_consistent_topology(self) -> None:
         top = self.atoms[0].root()
         for atom in self.atoms:
             current_top = atom.root()
+           
+            if isinstance(current_top, ResidueCap):
+                current_top = current_top.parent                
+
+            if isinstance(current_top, StructureRegion):
```

Comment:
In the current scheme, `StructureRegion` is the parent of the `ResidueCap`

Reply:
It is not right now right? I saw it is set to ghost parent.

Reply:
And I think it should not too logically. Because StructureRegion is not a real independent structure, it is under the context of a Structure() (it's topology). In this case, I think it is best to have ResidueCap have no parent and it can find the context by link_residue

Reply:
Makes sense! I will implement this change

Reply:
Removed it!